open_lm/model.py

```diff
@@ -222,14 +270,24 @@ def __init__(self, layer_id, args: Params):
             self.feed_forward = MoE(moe_args)
 
         self.layer_id = layer_id
-        self.attention_norm = args.norm_type(
-            args.dim,
-            eps=args.norm_eps,
-        )
-        self.ffn_norm = args.norm_type(
-            args.dim,
-            eps=args.norm_eps,
-        )
+        if using_te:
+            self.attention_norm = te.LayerNorm(
+                args.dim,
+                eps=args.norm_eps,
+            )
+            self.ffn_norm = te.LayerNorm(
+                args.dim,
+                eps=args.norm_eps,
+            )
+        else:
+            self.attention_norm = args.norm_type(
```

Comment:
can we just add te.LayerNorm as one of the `args.norm_type`?

Reply:
Added this in the latest commit based on presence of TE, TE.LayerNorm or NN.LayerNorm will be considered.